### PR TITLE
Mobx fix magda init

### PR DESCRIPTION
**Pull request by soyarsauce**
### What this PR does

Fixes https://github.com/TerriaJS/terriajs/issues/4833

Allow Magda backed maps to use an inline `terria-init` catalog without it getting overwritten by map-config before it can be parsed

Fixes https://github.com/TerriaJS/terriajs/issues/4834

Deprecates `proxyableDomainsUrl` config param

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

**Comment by soyarsauce** [comment]
Restarted job, seems like it got stuck on saucelabs

**Comment by soyarsauce** [comment]
Tests were actually broken - I even mentally noted to myself to uninstall the inlined stub but ended up not. Fixed them now!

File: lib/Models/Terria.ts [hunk 1]
```diff
@@ -876,16 +876,20 @@ export default class Terria {
     const configParams =
       aspects["terria-config"] && aspects["terria-config"].parameters;
 
-    configParams.initializationUrls =
-      aspects["terria-config"] && aspects["terria-config"].initializationUrls;
+    // TODO: what purpose does this serve? it's not a config parameter
+    // configParams.initializationUrls =
+    //   aspects["terria-config"] && aspects["terria-config"].initializationUrls;
```
Comment (rowanwins) [inline lib/Models/Terria.ts]: Agreed, I think this can be culled, from my reading of the code it serves no purpose.

**Review by rowanwins — APPROVED** [review]
Looks good

File: lib/Models/Terria.ts (hunk 1 above)
Comment (soyarsauce) [reply lib/Models/Terria.ts]: image

Thanks for confirming 😀

Comment (rowanwins) [reply lib/Models/Terria.ts]: I gitblame you @soyarsauce 😂 
Screen_2020_10_07

Comment (soyarsauce) [reply lib/Models/Terria.ts]: Hahaha sorry for misleading with half-pseudo-code!! You'll note my suggestion doesn't change the `initializationUrls` key on `configParams`. It looks like a naming problem, should have been called `initUrls` or something - `const initUrls = aspects["terria-config"] && aspects["terria-config"].initializationUrls;` and done what you will with the list of init urls there.

Anyway the extra mocked config map tests will help somewhat ensure things are loaded the right way 👍 it was quite difficult to verify without those unit tests.

**Pull request merged**